src/Features/LanguageServer/ProtocolUnitTests/Completion/CompletionTests.cs

```diff
@@ -192,28 +194,73 @@ class A
+            var document = solution.Projects.First().Documents.First();
+
+            // Set to use prototype completion behavior (i.e. feature flag).
+            var options = solution.Workspace.Options.WithChangedOption(CompletionOptions.UseLSPPrototypeBehavior, LanguageNames.CSharp, true);
+            Assert.True(solution.Workspace.TryApplyChanges(solution.WithOptions(options)));
+
+            var textEdit = GenerateTextEdit(@"\\A", startLine: 5, startChar: 19, endLine: 5, endChar: 19);
+
+            var expected = await CreateCompletionItemAsync(
+                label: @"\A", LSP.CompletionItemKind.Text, new string[] { "Text" }, completionParams, document, textEdit: textEdit,
+                sortText: "0000").ConfigureAwait(false);
+
+            var results = await RunGetCompletionsAsync(testLspServer, completionParams).ConfigureAwait(false);
+            AssertJsonEquals(expected, results.Items.First());
+        }
+
+        [Fact]
+        [WorkItem(50964, "https://github.com/dotnet/roslyn/issues/50964")]
+        public async Task TestGetRegexLiteralCompletionsAsync()
+        {
+            var markup =
+@"using System.Text.RegularExpressions;
+class A
+{
+    void M()
+    {
+        new Regex(@""\{|caret:|}"");
+    }
+}";
+            using var testLspServer = CreateTestLspServer(markup, out var locations);
+            var completionParams = CreateCompletionParams(
+                locations["caret"].Single(),
+                invokeKind: LSP.VSCompletionInvokeKind.Explicit,
                 triggerCharacter: "\\",
                 triggerKind: LSP.CompletionTriggerKind.TriggerCharacter);
 
-            var document = testLspServer.GetCurrentSolution().Projects.First().Documents.First();
+            var solution = testLspServer.GetCurrentSolution();
+            var document = solution.Projects.First().Documents.First();
+
+            // Set to use prototype completion behavior (i.e. feature flag).
+            var options = solution.Workspace.Options.WithChangedOption(CompletionOptions.UseLSPPrototypeBehavior, LanguageNames.CSharp, true);
+            Assert.True(solution.Workspace.TryApplyChanges(solution.WithOptions(options)));
+
+            var textEdit = GenerateTextEdit(@"\A", startLine: 5, startChar: 20, endLine: 5, endChar: 21);
 
             var expected = await CreateCompletionItemAsync(
-                "\\A", LSP.CompletionItemKind.Text, new string[] { "Text" }, completionParams, document, sortText: "0000").ConfigureAwait(false);
+                label: @"\A", LSP.CompletionItemKind.Text, new string[] { "Text" }, completionParams, document, textEdit: textEdit,
+                sortText: "0000").ConfigureAwait(false);
 
             var results = await RunGetCompletionsAsync(testLspServer, completionParams).ConfigureAwait(false);
             AssertJsonEquals(expected, results.Items.First());
         }
 
         [Fact]
-        [WorkItem(47743, "https://github.com/dotnet/roslyn/issues/47743")]
```

Comment:
This work item revolved around syntax highlighting and doesn't relate to this test at all, so I removed it.